src/concrete/ml/quantization/quantized_module.py

```diff
@@ -532,22 +558,30 @@ def quantize_input(self, *x: numpy.ndarray) -> Union[numpy.ndarray, Tuple[numpy.
 
         return q_x[0] if len(q_x) == 1 else q_x
 
-    def dequantize_output(self, q_y_preds: numpy.ndarray) -> numpy.ndarray:
+    def dequantize_output(
+        self, *q_y_preds: numpy.ndarray
+    ) -> Union[numpy.ndarray, Tuple[numpy.ndarray, ...],]:
         """Take the last layer q_out and use its de-quant function.
 
         Args:
-            q_y_preds (numpy.ndarray): Quantized output values of the last layer.
+            q_y_preds (numpy.ndarray): Quantized outputs values.
 
         Returns:
             numpy.ndarray: De-quantized output values of the last layer.
         """
-        y_preds = tuple(
-            output_quantizer.dequant(q_y_preds) for output_quantizer in self.output_quantizers
+        # Make sure that we have as many predictions as quantizers
+        assert len(q_y_preds) == len(self.output_quantizers), (
+            f"{len(q_y_preds)=} != {len(self.output_quantizers)=} "
+            "but the number of outputs to de-quantize should match the number of quantizers."
```

Comment:
small detail but I would maybe specificy ` the number of output quantizers` to be more precise